panther_analysis_tool/rule.py

```diff
@@ -289,55 +379,84 @@ def run(
             return rule_result
 
         try:
-            rule_result.title_output = self._get_title(event, use_default_on_exception=batch_mode)
+            rule_result.title_defined = self._is_function_defined("title")
+            rule_result.title_output = self._get_title(
+                event,
+                function_defined=rule_result.title_defined,
+                use_default_on_exception=batch_mode,
+            )
         except Exception as err:  # pylint: disable=broad-except
             rule_result.title_exception = err
 
         try:
+            rule_result.description_defined = self._is_function_defined(DESCRIPTION_FUNCTION)
```

Comment:
I used constants for the function names (which can be reused in `main` as well for the list of function names), to reduce the likelihood of an error or discrepancy between the two literals.